src/common/src/array/primitive_array.rs

```diff
@@ -244,15 +257,15 @@ impl<T: PrimitiveArrayItemType> ArrayBuilder for PrimitiveArrayBuilder<T> {
         }
     }
 
-    fn append(&mut self, value: Option<T>) {
+    fn append_n(&mut self, n: usize, value: Option<T>) {
         match value {
             Some(x) => {
-                self.bitmap.append(true);
-                self.data.push(x);
+                self.bitmap.append_n(n, true);
+                self.data.extend(std::iter::repeat(x).take(n));
```

Comment:
How's the performance compared to `itertools::repeat_n`? 🤔

Reply:
I did a quick bench. It turns out `itertools::repeat_n` is bad. 😇

```
std                     time:   [29.624 ns 29.846 ns 30.158 ns]
itertools               time:   [1.2989 µs 1.3006 µs 1.3028 µs]
```

```rust
use criterion::{criterion_group, criterion_main, Criterion};

criterion_group!(benches, bench_expr);
criterion_main!(benches);

fn bench_expr(c: &mut Criterion) {
    c.bench_function("std", |bencher| {
        bencher.iter(|| {
            let mut v = vec![];
            v.extend(std::iter::repeat(0u8).take(1024));
        })
    });
    c.bench_function("itertools", |bencher| {
        bencher.iter(|| {
            let mut v = vec![];
            v.extend(itertools::repeat_n(0u8, 1024));
        })
    });
}
```

Reply:
I guess the std library must be special optimized. 😄

Reply:
I modified some code of `itertools::repeat_n` and the time dropped from 1.3 us to 362.34 ns. 😇

Then I compared it with the code in std. It turns out the latter has `unsafe impl TrustedLen`. After adding it to itertools the performance became the same. 😇

Reply:
Looks like we found another chance to "water" a PR.

Reply:
You're right. 🥵 3rd party libraries cannot easily gain performance benefits from nightly features or built-in features (like `default fn`).